validmind/vm_models/test/metric_v2.py

```diff
@@ -0,0 +1,120 @@
+# Copyright © 2023-2024 ValidMind Inc. All rights reserved.
+# See the LICENSE file in the root of this repository for details.
+# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial
+
+"""
+Class for storing ValidMind metric objects and associated
+data for display and reporting purposes
+"""
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import ClassVar, Optional, Union
+
+import pandas as pd
+
+from validmind.vm_models import MetricResult
+
+from ...utils import clean_docstring
+from .result_wrapper import MetricResultWrapper
+from .test import Test
+
+
+@dataclass
+class UnitMetric(Test):
+    """
+    Metric objects track the schema supported by the ValidMind API
+    """
+
+    # Class Variables
+    test_type: ClassVar[str] = "UnitMetric"
+
+    type: ClassVar[str] = ""  # type of metric: "training", "evaluation", etc.
+    scope: ClassVar[str] = ""  # scope of metric: "training_dataset"
+    value_formatter: ClassVar[Optional[str]] = None  # "records" or "key_values"
+
+    # Instance Variables
+    result: MetricResultWrapper = None  # populated by cache_results() method
```

Comment:
For now we can just store the results as a `MetricResult`. That will allow removing a dependency on `MetricResultWrapper` which is not necessary in this case since we're not showing unit metric outputs in the suite results.

```suggestion
    result: MetricResult = None  # populated by cache_results() method
```

Reply:
Done.